kickstarter/android-oss
# NT-1259: Firebase Analytics dependency

# 📲 What

- Added Firebase Analytics Kotlin dependency.
- Updated Firebase Messaging (it's also a dependency for Firebase analytics, so I'm force to use the same version as them)
- Updated some gems 
-- Rubyzip & excon to fix the vulnerability issue in the repo
-- The rest of them needed for fastlane plugin

# 🤔 Why
To be able to track more data from Android app.
In that capture as you can see no crash free user information and hitting on the lastest release button just gave us information from the las one available one 1.18.1
![Screen Shot 2020-05-19 at 11 04 02 AM](https://user-images.githubusercontent.com/4083656/82509332-6eb36080-9abc-11ea-9dcd-b0bc832bf833.png)

# 👀 See

Now we have our data, in internal release for now 😆 !!
![Screen Shot 2020-05-20 at 5 17 31 PM](https://user-images.githubusercontent.com/4083656/82509828-d74f0d00-9abd-11ea-9630-3c61d1cf225c.png)
![Screen Shot 2020-05-20 at 2 29 39 PM](https://user-images.githubusercontent.com/4083656/82509748-a078f700-9abd-11ea-997b-7ee6410b08e4.png)
* Including the app version adoption, which will help us decide the best moment for updating rollouts 
![Screen Shot 2020-05-20 at 5 27 15 PM](https://user-images.githubusercontent.com/4083656/82510420-4f6a0280-9abf-11ea-9b72-898c01d2fb6f.png)

# 📋 QA
- Execute in your environment the next scripts and nothing should fail:
```bundle install```
```bundle exec fastlane external|internal ``` --> depending on the type you wanna build
```./milkrun update-strings```
- Try to message someone using the app


# Story 📖

[Analytics dependency](https://kickstarter.atlassian.net/browse/NT-1259)
